ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -161,6 +170,31 @@ func (scbuilder *SiteConfigBuilder) getClusterCR(clusterId int, siteConfigTemp S
 	return mapIntf, nil
 }
 
+func (scbuilder *SiteConfigBuilder) getBiosConfigMapCR(clusterId int, siteConfigTemp SiteConfig, nodeId int) (interface{}, error) {
+
+	// if BiosConfigMapRef is not set for the host, use the one set for the cluster
+	if siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].BiosConfigMapRef.Name == "" {
+		siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].BiosConfigMapRef = siteConfigTemp.Spec.Clusters[clusterId].BiosConfigMapRef
+	}
+
+	annotations := map[string]string{"argocd.argoproj.io/sync-wave": "1"}
+	metadata := Metadata{Name: siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].HostName,
+		Namespace: siteConfigTemp.Spec.Clusters[clusterId].ClusterName, Annotations: annotations}
+	biosCM := ConfigMap{"v1", "ConfigMap", metadata, make(map[string]string)}
+
+	biosFile := siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].BiosConfigMapRef.Name
+	fileData, err := ReadFile(biosFile)
+	if err != nil {
+		return biosCM, err
+	}
+	err = yaml.Unmarshal(fileData, biosCM.Data)
+	if err != nil {
+		return biosCM, err
+	}
+	siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].BiosConfigMapRef.Name = siteConfigTemp.Spec.Clusters[clusterId].Nodes[nodeId].HostName
```

Comment:
I am thinking using a host name as the generated config map name is not ideal as it could clash with the extra manifest config map, when the host name and cluster are the same.  May we ask the user to enter a config map name or generate a unique name for each host?

Reply:
yes, let the user enter it